src/core/qgsmaplayerlegend.cpp

```diff
@@ -359,6 +360,19 @@ QList<QgsLayerTreeModelLegendNode *> QgsDefaultVectorLayerLegend::createLayerTre
 {
   QList<QgsLayerTreeModelLegendNode *> nodes;
 
+  if ( mLayer )
+  {
+    QString placeholderImage = mLayer->legendPlaceholderImage();
+    if ( !placeholderImage.isEmpty() )
+    {
+      QgsImageCache ic;
```

Comment:
This should use the image cache instance attached to QgsApplication

Reply:
Ah cool, didn't know there is an image cache instance in QgsApplication